albumentations/augmentations/functional.py

```diff
@@ -381,10 +381,19 @@ def _shift_hsv_non_uint8(img, hue_shift, sat_shift, val_shift):
 
 
 def shift_hsv(img, hue_shift, sat_shift, val_shift):
+    is_gray = is_grayscale_image(img)
+    if is_gray:
+        img = cv2.cvtColor(img, cv2.COLOR_GRAY2RGB)
```

Comment:
I think it's better to shift only value for grayscale and throw exception in we want to shift other things because hsv shift does not make sence to me

Reply:
> hsv shift does not make sence to me

I agree with you.

But question is what we want from all transforms.

In dataset must be all RGB images, so exception will allow to catch error.
But what if we know, that dataset has few grayscale images and we need shift_hsv transform.

I think is the best solutions is add warning that will be say, that something wrong, but transform will be executed.

Reply:
agree, so let's just shift V(value) and ignore other and warn that hue and saturation are not applicable to grayscale if they are not zeros.